### PR TITLE
Add category

### PR DESCRIPTION
**Pull request by johannaanden**
### PT Link
[PT Link](https://www.pivotaltracker.com/story/show/162193281)

### User Story
As a visitor,
In order to navigate the products
I would like to see the products sorted in categories

### Actions
- Install faker gem
- Updated seeds file
- Create a new table, category
- Add associations between models 
- Add foreign key
- DB migrations
- DB setup for Faker

### LoFi
Visit PT to see the LoFi
[PT Link](https://www.pivotaltracker.com/story/show/162193281)

### Screenshot
screen shot 2018-11-30 at 11 22 39

### Gems added
- Faker

### What have we learned
- Using faker gem
- Using seeds file
- How to create a new table
- How to add associations
- How to add foreign key

### CODE REVIEW

File: spec/factories/categories.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :category do
+    category "MyString"
```
Comment (faraznaeem) [inline spec/factories/categories.rb]: please be consistent use `{ }`

File: features/menu.feature [hunk 2]
```diff
@@ -3,14 +3,42 @@ Feature: User can see menu
...
-        And I see '50'
\ No newline at end of file
+        And I see '50'
+        And I see 'Starter'
+    
+    Scenario: User can navigate the products
+        And I click "Login"
+        When I fill in 'Email' with 'Don@trump.com'
+        And I fill in 'Password' with 'maga2020'
+        And I click 'Log in'
```
Comment (faraznaeem) [inline features/menu.feature]: Please refactor the login steps to `Given the user is logged in`

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -4,7 +4,16 @@
 
 Given('the following products exists') do |table|
     table.hashes.each do |product|
-        FactoryBot.create(:product, product)
+        category = Category.find_by(category: product[:category])
+        # binding.pry
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: remove

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: spec/factories/categories.rb (hunk 1 above)
Comment (johannaanden) [reply spec/factories/categories.rb]: done

File: features/menu.feature (hunk 2 above)
Comment (johannaanden) [reply features/menu.feature]: Changed to: And I have logged in

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (johannaanden) [reply features/step_definitions/basic_steps.rb]: done

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**